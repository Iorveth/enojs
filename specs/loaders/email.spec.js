const eno = require('enojs');

const examples = {
  'john.doe@eno-lang.org': 'john.doe@eno-lang.org',
  'john.doe@eno-lang':     false,
  '@eno-lang.org':         false,
  'john.doe@.org':         false
};

describe('email loader', () => {
  for(let [value, result] of Object.entries(examples)) {
    describe(value, () => {
      const document = eno.parse(`value: ${value}`);

      if(result === false) {
        it('throws an error', () => {
          expect(() => document.email('value')).toThrowErrorMatchingSnapshot();
        });
      } else {
        it(`returns ${result}`, () => {
          expect(document.email('value')).toBe(result);
        });
      }
    });
  }
});
