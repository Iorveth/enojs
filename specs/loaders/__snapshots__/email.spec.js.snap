// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`email @eno-lang.org throws an error 1`] = `
"'email' must contain a valid email address, for instance 'jane.doe@eno-lang.org'.

   Line | Content
 >    1 | email: @eno-lang.org
"
`;

exports[`email john.doe@.org throws an error 1`] = `
"'email' must contain a valid email address, for instance 'jane.doe@eno-lang.org'.

   Line | Content
 >    1 | email: john.doe@.org
"
`;

exports[`email john.doe@eno-lang throws an error 1`] = `
"'email' must contain a valid email address, for instance 'jane.doe@eno-lang.org'.

   Line | Content
 >    1 | email: john.doe@eno-lang
"
`;
